### PR TITLE
ON-7752 # Added functions for managing username/password

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -24,8 +24,15 @@ declare namespace authService {
   function setFormsKeyToken(formsKeyToken: string): void
   function isAuthorised(formsAppId: number): Promise<boolean>
   function requestAccess(formsAppId: number): Promise<void>
-  function login(): Promise<string>
+  function loginHostedUI(identityProvider: string | void): Promise<string>
```
Comment (Zaxist) [inline index.d.ts:27]:
```suggestion
  function loginHostedUI(identityProvide?r: string): Promise<string>
```

File: index.d.ts [hunk 2]
```diff
@@ -24,8 +24,15 @@ declare namespace authService {
...
   function handleAuthentication(): Promise<string>
+  function loginUsernamePassword(): Promise<
+    ((newPassword: string) => Promise<void>) | void
+  >
+  function changePassword(): Promise<void>
```
Comment (Zaxist) [inline index.d.ts:32]: This should new and old passwords right?

Comment (mymattcarroll) [reply index.d.ts:32]: yep

File: index.d.ts (hunk 1 above)
Comment (Zaxist) [reply index.d.ts:27]: Actually, can we use an enum of the possible options?

File: index.d.ts [hunk 3]
```diff
@@ -24,8 +24,15 @@ declare namespace authService {
   function setFormsKeyToken(formsKeyToken: string): void
   function isAuthorised(formsAppId: number): Promise<boolean>
   function requestAccess(formsAppId: number): Promise<void>
-  function login(): Promise<string>
+  function loginHostedUI(identityProvider: string | void): Promise<string>
   function handleAuthentication(): Promise<string>
+  function loginUsernamePassword(): Promise<
```
Comment (Zaxist) [inline index.d.ts:29]: Missing args

File: index.d.ts [hunk 4]
```diff
@@ -24,8 +24,15 @@ declare namespace authService {
...
+    ((newPassword: string) => Promise<void>) | void
+  >
+  function changePassword(): Promise<void>
+  function forgotPassword(): Promise<
```
Comment (Zaxist) [inline index.d.ts:33]: This takes an arg too

**Comment by Zaxist** [comment]
`init()` function type also needs updating

File: index.d.ts (hunk 1 above)
Comment (mymattcarroll) [reply index.d.ts]: No we can't SAML identity providers are custom strings.

**Pull request merged**